modules/core/03-connection/keeper/verify.go

```diff
@@ -218,20 +219,49 @@ func (k Keeper) VerifyPacketCommitment(
 		return errorsmod.Wrapf(clienttypes.ErrClientNotActive, "client (%s) status is %s", clientID, status)
 	}
 
-	// get time and block delays
-	timeDelay := connection.GetDelayPeriod()
-	blockDelay := k.getBlockDelay(ctx, connection)
+	var multihopProof commitmenttypes.MsgMultihopProofs
+	if len(connectionHops) > 1 {
+		if err := k.cdc.Unmarshal(proof, &multihopProof); err != nil {
+			return errorsmod.Wrap(err, "failed to unmarshal multi-hop proof")
+		}
+	} else {
+		multihopProof.KeyProof = &commitmenttypes.MultihopProof{}
+		multihopProof.KeyProof.Proof = proof
+	}
+
+	// get maxmimum delay for any hop in the channel path
+	delayTimePeriod, err := k.GetMaximumDelayPeriod(connection, &multihopProof)
+	if err != nil {
+		return err
+	}
+	// FIXME: this may not be accurate for multi-hop channels
+	expectedTimePerBlock := k.GetParams(ctx).MaxExpectedTimePerBlock
+
+	// get the last hop connection on the other side of the multihop channel
+	// the last hop connection is the connection end on the chain before the counterparty multihop chain
+	connectionEnd, err := k.GetLastHopConnectionEnd(connection, &multihopProof)
+	if err != nil {
+		return errorsmod.Wrap(err, "failed to get connection end")
+	}
 
+	prefix := connectionEnd.GetCounterparty().GetPrefix()
 	merklePath := commitmenttypes.NewMerklePath(host.PacketCommitmentPath(portID, channelID, sequence))
-	merklePath, err = commitmenttypes.ApplyPrefix(connection.GetCounterparty().GetPrefix(), merklePath)
+	merklePath, err = commitmenttypes.ApplyPrefix(prefix, merklePath)
 	if err != nil {
 		return err
 	}
 
-	if err := clientState.VerifyMembership(
-		ctx, clientStore, k.cdc, height,
-		timeDelay, blockDelay,
-		proof, merklePath, commitmentBytes,
+	if err := clientState.VerifyMultihopMembership(
```

Comment:
Verify multihop proof membership should work for either a single hop proof converted to multihop as well.